android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -49,15 +49,15 @@ private const val VIEW_HOLDER_TYPE_ADDON = 2
  * an add-on such as recommended, unsupported or installed. In addition, it will perform actions
  * such as installing an add-on.
  *
- * @property addonCollectionProvider Provider of AMO collection API.
+ * @property addonsProvider Provider of AMO collection API.
  * @property addonsManagerDelegate Delegate that will provides method for handling the add-on items.
  * @param addons The list of add-on based on the AMO store.
  * @property style Indicates how items should look like.
  * @property excludedAddonIDs The list of add-on IDs to be excluded from the recommended section.
  */
 @Suppress("LargeClass")
 class AddonsManagerAdapter(
-    private val addonCollectionProvider: AddonCollectionProvider,
+    private val addonsProvider: DefaultAddonsProvider,
```

Comment:
@Amejia481 Why would we inject the concrete implementation here too?

Reply:
Thanks for point this, unfortunately same as https://github.com/mozilla-mobile/firefox-android/pull/3154#discussion_r1287670763